tests/test_session_1_byref_false.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
+
+# Author: Leonardo Gama (@leogama)
+# Copyright (c) 2008-2016 California Institute of Technology.
+# Copyright (c) 2016-2022 The Uncertainty Quantification Foundation.
+# License: 3-clause BSD.  The full license text is available at:
+#  - https://github.com/uqfoundation/dill/blob/master/LICENSE
+
+from __future__ import print_function
+import dill, os, sys
+
+session_file = 'session-byref-%s.pkl'
```

Comment:
this requires `test_session_0.py` to be run first.  instead you could import and generate the `*pkl` file if it doesn't exist.  I don't like the interdependency between these tests. Each test should be independently runnable, regardless of the order executed.

Reply:
This won't work because for generating the `.pkl` file it will need to load all the modules necessary to populate the namespace. The idea behind saving on one file and loading in the next is to faithfully reproduce the use case of loading the dumped file in a fresh new session.

Also, _there are hidden spirits in the importing machinery seeing everything you do and playing tricks._ I tried to run all the tests in a single file at first but hit a weird bug where, after cleaning the `sys.modules` dictionary for the second time, the `calendar` module wouldn't show up there again, neither using the `__import__()` function nor the `import` statement and variants.

If you are convinced about having independent tests, the alternative I see is to use a python subprocess to test the file loading and module importing in a fresh interpreter session. Then the test files could be merged again.

Reply:
The subprocess solution worked flawlessly! It went out very clean and elegant in a single file (I was expecting to end with some macaroni code).